src/tcgwars/logic/impl/gen3/LegendMaker.groovy

```diff
@@ -861,14 +861,12 @@ public enum LegendMaker implements LogicCardInfo {
         weakness P
         pokeBody "Sol Shade", {
           text "As long as you have Solrock in play, each player's [R] Pokémon (excluding Pokémon-ex) can't use any Poké-Powers."
-          delayedA {
             getterA (IS_ABILITY_BLOCKED) { Holder h->
-              if (my.all.find{it.name == 'Solrock'} && h.effect.target.types.contains(R) && !h.effect.target.EX && h.effect.ability instanceof PokePower) {
+              if (my.all.find{it.name == 'Solrock'} && h.effect.target.types.contains(R) && !h.effect.target.topPokemonCard.cardTypes.is(EX) && h.effect.ability instanceof PokePower) {
```

Comment:
.EX has the same functionality as topPokemonCard.cardTypes.is(EX). I prefer the the former as it keeps line length down but either is acceptable